File: core/tests/fixtures/mothers/provider_mother.py
from src.domain.provider import Provider
from tests.fixtures.mothers import EnterpriseContactMother, AddressMother
from tests.fixtures.mothers.base import BaseMother


class ProviderMother(BaseMother):
    """Mother class for creating Provider instances for testing purposes."""

    _AMAZON = {
        'name': 'Amazon',
        'comercial_name': 'Amazon, Inc.',
        'contact': EnterpriseContactMother.amazon,
        'address': AddressMother.anytown,
        'vat_number': 'B36565656',
        'created_by': 'test',
    }

    _BEST_BUY = {
        'name': 'Best Buy',
        'comercial_name': 'Best Buy Co., Inc.',
        'contact': EnterpriseContactMother.best_buy,
        'address': AddressMother.anytown,
        'vat_number': 'B36565656',
        'created_by': 'test',
    }

    @classmethod
    def amazon(cls, **overrides) -> Provider:
        """Returns a generic provider instance."""
        return cls._create(cls._AMAZON, overrides)

    @classmethod
    def best_buy(cls, **overrides) -> Provider:
        """Returns a generic provider instance."""
        return cls._create(cls._BEST_BUY, overrides)

    @classmethod
    def _create(cls, values: dict, overrides: dict) -> Provider:
        """Creates a Provider instance with the given values and overrides."""
        data = cls._merge(values, overrides)
        return Provider(**data)

File: editor/tests/fixtures/mothers/provider.mother.ts
import { EnterpriseContactMother } from '@tests/fixtures/mothers/enterprise-contact.mother';
import { AddressMother } from '@tests/fixtures/mothers/address.mother';
import { BaseMother } from '@tests/fixtures/mothers/base/base-mother';
import { Provider } from '@/providers/model/provider';
import { MotherMapping } from '@tests/fixtures/mothers/base/mother-mapping';

const AMAZON = {
  id: 'b684646e-79cd-410f-a601-08c2e49e80ed',
  name: 'Amazon',
  comercialName: 'Amazon, Inc.',
  contact: EnterpriseContactMother.amazon,
  address: AddressMother.anytown,
  vatNumber: 'B36565656',
  createdBy: 'test',
  createdAt: ['2026-04-28T10:25:36', MotherMapping.DATE],
};

const BEST_BUY = {
  id: '68228309-8323-4122-9557-fe7af05ce846',
  name: 'Best Buy',
  comercialName: 'Best Buy Co., Inc.',
  contact: EnterpriseContactMother.bestBuy,
  address: AddressMother.anytown,
  vatNumber: 'B36565656',
  createdBy: 'test',
  createdAt: ['2026-04-28T10:25:36', MotherMapping.DATE],
};

export type ProviderFixture = typeof AMAZON;

export class ProviderMother extends BaseMother {
  static amazon(this: void, overrides?: Partial<Provider>): Provider {
    return ProviderMother.create(AMAZON, overrides);
  }

  static bestBuy(this: void, overrides?: Partial<Provider>): Provider {
    return ProviderMother.create(BEST_BUY, overrides);
  }

  protected static create(
    values: ProviderFixture,
    overrides?: Partial<Provider>,
  ): Provider {
    const fields = ProviderMother.merge<ProviderFixture, Provider>(
      values,
      overrides,
    );

    return Provider.from(fields);
  }
}
